modules/internal/localeModels/flat/LocaleModel.chpl

```diff
@@ -286,32 +286,44 @@ module LocaleModel {
   private extern proc chpl_memhook_md_num(): chpl_mem_descInt_t;
 
   // The allocator pragma is used by scalar replacement.
+
+  // Note that there's 2 chpl_here_alloc() procs. This one explicitly takes an
+  // int(64) and is marked with "locale model alloc" while the one below takes
+  // a generic `integral`. This one is used/called by the compiler and so its
+  // type must be fully defined.
   pragma "allocator"
   pragma "locale model alloc"
```

Comment:
That looks good as far as it goes, but maybe add something like 'There must be precisely one function marked with "locale model alloc" in each compilation. Ordinarily this function is in the top-level locale model.'